### PR TITLE
VectorCombine: lift one-use limitation in foldExtractedCmps

### PR DESCRIPTION
**Pull request by artagnon**
There are artificial one-use limitations on foldExtractedCmps. Adjust the costs to account for multi-use, and strip the one-use matcher, lifting the limitations.

### CODE REVIEW

**Comment by artagnon** [comment]
Gentle ping. Seems pretty straightforward?

File: llvm/lib/Transforms/Vectorize/VectorCombine.cpp [hunk 1]
```diff
@@ -1096,6 +1095,7 @@ bool VectorCombine::foldExtractedCmps(Instruction &I) {
                                 ShufMask);
   NewCost += TTI.getArithmeticInstrCost(I.getOpcode(), CmpTy);
   NewCost += TTI.getVectorInstrCost(*Ext0, CmpTy, CostKind, CheapIndex);
+  NewCost += !Ext0->hasOneUse() * Ext0Cost + !Ext1->hasOneUse() * Ext1Cost;
```
Comment (RKSimon) [inline llvm/lib/Transforms/Vectorize/VectorCombine.cpp:1098]: I'm not a fan of using bool in arithmetic tbh

Comment (RKSimon) [reply llvm/lib/Transforms/Vectorize/VectorCombine.cpp:1098]:
```
NewCost += Ext0->hasOneUse() ? 0 : Ext0Cost;
NewCost += Ext1->hasOneUse() ? 0 : Ext1Cost;

```

**Review by RKSimon — APPROVED** [review]
LGTM with one minor (and I tweaked the PR title to mention foldExtractedCmps)

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

**Pull request merged**